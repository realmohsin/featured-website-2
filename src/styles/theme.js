import { createMuiTheme } from '@material-ui/core/styles'

const primaryColor = '#0E5C60'
const primaryDarkColor = '#002B2E'
const primaryVariant1 = '#00464A'
const secondaryColor = '#CCAB63'
const goldVariant1 = '#E1C079'
const facebookColor = '#3C5B9B'
const googleStarColor = '#E7711B'
const fadedTextColor = '#94B7B8'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryColor,
      dark: primaryDarkColor
    },
    secondary: {
      main: secondaryColor
    },
    common: {
      primaryVariant1,
      goldVariant1,
      facebookColor,
      googleStarColor,
      fadedTextColor
    }
  },
  typography: {
    htmlFontSize: 10,
    fontFamily: ['Roboto', 'sans-serif'].join(', '),
    body1: {
      fontSize: '1.6rem',
      '@media (max-width: 600px)': {
        fontSize: '1.4rem'
      }
    },
    body2: {
      fontSize: '1.4rem',
      lineHeight: '1.5'
    }
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '*, *::before, *::after': {
          margin: 0,
          padding: 0
        },
        html: {
          fontSize: '62.5%'
        },
        a: {
          display: 'block',
          textDecoration: 'none',
          color: 'inherit'
        },
        img: {
          display: 'block'
        },
        'input, button, textarea, select': {
          display: 'block',
          font: 'inherit'
        },
        ul: {
          listStyle: 'none'
        },
        '::selection': {
          backgroundColor: primaryColor,
          color: 'white'
        },
        'a, button': {
          cursor: 'pointer'
        }
      }
    },
    MuiTypography: {
      gutterBottom: {
        marginBottom: '1.2em'
      }
    }
  },
  custom: {
    title1: {
      fontFamily: ['Piazzolla', 'serif'].join(', '),
      fontSize: '5.1rem',
      color: primaryColor,
      '@media (max-width: 600px)': {
        fontSize: '2rem'
      }
    },
    title2: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '4.6rem',
      color: primaryColor,
      fontWeight: 'normal',
      marginBottom: '8rem',
      textAlign: 'center'
    },
    title3: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '3.4rem',
      color: secondaryColor,
      fontWeight: 'normal'
    },
    title4: {
      fontSize: '2.2rem',
      color: primaryColor,
      textTransform: 'uppercase',
      wordSpacing: '0.5px'
    },
    title5: {
      fontSize: '1.5rem',
      color: secondaryColor,
      textTransform: 'uppercase',
      fontWeight: 'normal'
    },
    outlineTitle: {
      textShadow: `-1px 0 ${primaryColor}, 0 1px ${primaryColor}, 1px 0 ${primaryColor}, 0 -1px ${primaryColor}`
    },
    inSideDrawer: {
      transform: 'translateY(0)',
      opacity: 1,
      pointerEvents: 'auto',
      width: 'auto',
      height: 'auto',
      border: 'none',
      backgroundColor: 'white',
      boxShadow: 'inset 0 0 5px 2px rgba(0,0,0,.2)',
      '& a': {
        color: primaryColor,
        fontSize: '1.1rem',
        display: 'flex',
        height: '4rem',
        borderBottom: `1px solid #eee`,
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingRight: '1rem',
        fontWeight: 'normal',
        paddingLeft: '3.6rem'
      }
    }
  }
})

export default theme
