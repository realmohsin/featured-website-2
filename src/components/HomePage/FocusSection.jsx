import React, { useState, useEffect } from 'react'
import { graphql, useStaticQuery } from 'gatsby'
import { StaticImage, getImage } from 'gatsby-plugin-image'
import { makeStyles, useTheme } from '@material-ui/core/styles'
import useMediaQuery from '@material-ui/core/useMediaQuery'
import { Grid, Container, Typography } from '@material-ui/core'
import Button from '../Button'

const useStyles = makeStyles(theme => ({
  example: {
    // ...mobile first styles,
    [theme.breakpoints.up('sm')]: {
      //...sm and up styles
    },
    [theme.breakpoints.up('md')]: {
      //...md and up styles
    },
    '@media (min-width: 1320px)': {
      //...rules for above 1320px
    }
  },
  focusSection: {
    padding: '12rem 0'
  },
  focusTitle: {
    ...theme.custom.title2
  },
  row: {
    marginBottom: '5rem'
  },
  rowContent: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    padding: '0 5rem'
  },

  rowTitle: {
    ...theme.custom.title4,
    textAlign: 'center',
    marginBottom: '4rem'
  },
  divider: {
    width: '100%',
    height: '2px',
    backgroundColor: theme.palette.secondary.main
  },
  rowParagraph: {
    margin: '4rem 0 8rem'
  },
  button: {
    margin: '0 auto'
  },
  rowImgContainer: {
    padding: '2.3rem'
  }
}))

const FocusSection = props => {
  const classes = useStyles()
  // const data = useStaticQuery(graphql`
  //   query {
  //     focus1: file(name: { eq: "focus-1" }) {
  //       childImageSharp {
  //         fluid(maxWidth: 1275, maxHeight: 828) {
  //           ...GatsbyImageSharpFluid
  //         }
  //       }
  //     }
  //     focus2: file(name: { eq: "focus-2" }) {
  //       childImageSharp {
  //         fluid(maxWidth: 1275, maxHeight: 828) {
  //           ...GatsbyImageSharpFluid
  //         }
  //       }
  //     }
  //     focus3: file(name: { eq: "focus-3" }) {
  //       childImageSharp {
  //         fluid(maxWidth: 1275, maxHeight: 828) {
  //           ...GatsbyImageSharpFluid
  //         }
  //       }
  //     }
  //   }
  // `)

  return (
    <section className={classes.focusSection}>
      <Container>
        <h2 className={classes.focusTitle}>Our Focus</h2>
        <Grid container className={classes.row}>
          <Grid item xs={12} sm={5} className={classes.rowContent}>
            <h3 className={classes.rowTitle}>Business Law</h3>
            <div className={classes.divider} />
            <Typography variant='body1' className={classes.rowParagraph}>
              From startups to larger corporations, we provide a range of legal
              services to help your business grow.
            </Typography>
            <Button isLink text='Learn More' extraClassName={classes.button} />
          </Grid>
          <Grid item xs={12} sm={7} className={classes.rowImgContainer}>
            {/* <Img fluid={data.focus1.childImageSharp.fluid} alt='Business Law' /> */}
            <StaticImage
              src='../../assets/images/focus-1.jpg'
              alt='Business Law'
            />
          </Grid>
        </Grid>

        <Grid container className={classes.row}>
          <Grid item xs={12} sm={7} className={classes.rowImgContainer}>
            {/* <Img
              fluid={data.focus2.childImageSharp.fluid}
              alt='Entertainment Law'
            /> */}
            <StaticImage
              src='../../assets/images/focus-2.jpg'
              alt='Entertainment Law'
            />
          </Grid>

          <Grid item xs={12} sm={5} className={classes.rowContent}>
            <h3 className={classes.rowTitle}>Entertainment Law</h3>
            <div className={classes.divider} />
            <Typography variant='body1' className={classes.rowParagraph}>
              Weâ€™re here to provide guidance to production companies, producers,
              talent and other industry professionals as you build your career.
            </Typography>
            <Button isLink text='Learn More' extraClassName={classes.button} />
          </Grid>
        </Grid>

        <Grid container className={classes.row}>
          <Grid item xs={12} sm={5} className={classes.rowContent}>
            <h3 className={classes.rowTitle}>Business Disputes</h3>
            <div className={classes.divider} />
            <Typography variant='body1' className={classes.rowParagraph}>
              Our experienced team of attorneys can help you navigate a
              disagreement, business divorce or recover a debt. We can assist
              with the litigation or defense of commercial matters in state,
              federal bankruptcy, federal district and federal appellate courts.
            </Typography>
            <Button isLink text='Learn More' extraClassName={classes.button} />
          </Grid>
          <Grid item xs={12} sm={7} className={classes.rowImgContainer}>
            <StaticImage
              src='../../assets/images/focus-3.jpg'
              alt='Business Disputes'
            />
          </Grid>
        </Grid>
      </Container>
    </section>
  )
}

export default FocusSection
