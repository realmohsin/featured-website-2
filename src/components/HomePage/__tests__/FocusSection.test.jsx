import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core/styles";
import { createMuiTheme } from "@material-ui/core/styles";
import FocusSection from "../FocusSection";

// Mock gatsby-plugin-image
jest.mock("gatsby-plugin-image", () => ({
  StaticImage: ({ src, alt, className, ...props }) => (
    <img src={src} alt={alt} className={className} {...props} />
  ),
  getImage: jest.fn((image) => image)
}));

// Mock the Button component
jest.mock("../../Button", () => ({
  __esModule: true,
  default: ({ text, href, ...props }) => (
    <a href={href} {...props}>
      {text}
    </a>
  ),
}));

const theme = createMuiTheme({
  palette: {
    primary: { main: "#0E5C60", light: "#4A8B8F" },
    secondary: { main: "#CCAB63" },
  },
  breakpoints: {
    down: () => "@media (max-width: 600px)",
  },
  custom: {
    title2: {
      fontSize: "5.6rem",
      fontWeight: "bold",
    },
    title3: {
      fontSize: "3.2rem",
    },
    title5: {
      fontSize: "1.8rem",
    },
  },
});

const renderWithTheme = (component) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};

describe("FocusSection Component", () => {
  it("renders focus section with all content", () => {
    const { container } = renderWithTheme(<FocusSection />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
