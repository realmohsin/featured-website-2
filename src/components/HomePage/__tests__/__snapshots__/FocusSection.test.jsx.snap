// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusSection Component renders focus section with all content 1`] = `
<section
  class="makeStyles-focusSection-1"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h2
      class="makeStyles-focusTitle-2"
    >
      Our Focus
    </h2>
    <div
      class="MuiGrid-root makeStyles-row-3 MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-rowContent-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
      >
        <h3
          class="makeStyles-rowTitle-5"
        >
          Business Law
        </h3>
        <div
          class="makeStyles-divider-6"
        />
        <p
          class="MuiTypography-root makeStyles-rowParagraph-7 MuiTypography-body1"
        >
          From startups to larger corporations, we provide a range of legal services to help your business grow.
        </p>
      </div>
      <div
        class="MuiGrid-root makeStyles-rowImgContainer-9 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
      >
        <img
          alt="Business Law"
          src="../../assets/images/focus-1.jpg"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-row-3 MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-rowImgContainer-9 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
      >
        <img
          alt="Entertainment Law"
          src="../../assets/images/focus-2.jpg"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-row-3 MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-rowContent-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
      >
        <h3
          class="makeStyles-rowTitle-5"
        >
          Business Disputes
        </h3>
        <div
          class="makeStyles-divider-6"
        />
        <p
          class="MuiTypography-root makeStyles-rowParagraph-7 MuiTypography-body1"
        >
          Our experienced team of attorneys can help you navigate a disagreement, business divorce or recover a debt. We can assist with the litigation or defense of commercial matters in state, federal bankruptcy, federal district and federal appellate courts.
        </p>
      </div>
      <div
        class="MuiGrid-root makeStyles-rowImgContainer-9 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
      >
        <img
          alt="Business Disputes"
          src="../../assets/images/focus-3.jpg"
        />
      </div>
    </div>
  </div>
</section>
`;
